### PR TITLE
Fix MapActivity crash while requesting for permissions

### PR DESCRIPTION
**Pull request by ekigamba**
1. Reading local styles from storage leads to crash while requesting for the permission
2. MapBox crashes due to invalid AccessToken if it is initialized while waiting for permissions

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 1]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
-                    maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND)) {
+                topLeftBound = bundle.getParcelable(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND);
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
-                    minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
+                if (bundle.containsKey(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND)) {
+                    bottomRightBound = bundle.getParcelable(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Can we not use MapBox's bounds for the vector layer? https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-vector-bounds

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 2]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
                 }
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
-                    cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
+                maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Can we not use MapBox's maxZoom property for the  vector layer? https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-vector-maxzoom

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 3]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_ZOOM)) {
-                    cameraZoom = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
+                minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Same here https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-vector-minzoom

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 4]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TILT)) {
-                    cameraTilt = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_TILT);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
+                cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: What about this https://www.mapbox.com/mapbox-gl-js/style-spec/#root-center

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 5]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_BEARING)) {
-                    cameraBearing = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_BEARING);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_ZOOM)) {
+                cameraZoom = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_ZOOM);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: What about https://www.mapbox.com/mapbox-gl-js/style-spec/#root-zoom

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 6]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                            sortFields = extractSortFields(mapboxStyleJSON);
-                            dataLayers = extractSourceNames(mapboxStyleJSON, sortFields);
-                            featuresMap = extractLayerData(mapboxStyleJSON, dataLayers);
-                            featuresMap = sortData(featuresMap, sortFields);
-                            displayInitialFeatures(featuresMap);
-                        } catch (JSONException e) {
-                            Log.e(TAG, Log.getStackTraceString(e));
-                        }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TILT)) {
+                cameraTilt = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_TILT);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: What about https://www.mapbox.com/mapbox-gl-js/style-spec/#root-pitch

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 7]
```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
+
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_BEARING)) {
+                cameraBearing = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_BEARING);
```
Comment (jasonrogena) [inline library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: What about https://www.mapbox.com/mapbox-gl-js/style-spec/#root-bearing

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 1 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I think vector layer zoom constraints and map zoom constraints are different

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 2 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I think vector layer zoom constraints and map zoom constraints are different

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 3 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I think vector layer zoom constraints and map zoom constraints are different

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 4 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I will look at this

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 5 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I will look at this

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 6 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I will look at this

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 7 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: I will look at this

**Comment by jasonrogena** [comment]
@ekigamba The point not defining variables that have already been defined in the Style Spec.

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 4 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: We can add this as an issue ❗️ ❗️

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 6 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Add this also as an issue ❗️ ❗️

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 7 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Add this also as an issue ❗️ ❗️

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 5 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Add this also as an issue ❗️ ❗️

**Comment by ekigamba** [comment]
Cool

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java (hunk 1 above)
Comment (jasonrogena) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: Going to create an issue for this.

**Pull request merged**